src/app/profile/info-form/info-form.component.ts

```diff
@@ -1,33 +1,46 @@
-import { Component } from '@angular/core';
-import {MatSelectModule} from '@angular/material/select';
-import {MatInputModule} from '@angular/material/input';
-import {MatFormFieldModule} from '@angular/material/form-field';
-import {MatDatepickerModule} from '@angular/material/datepicker';
-import {MatNativeDateModule} from '@angular/material/core';
+import { Component, Input, OnInit, Output, EventEmitter, OnChanges, SimpleChanges } from '@angular/core';
+import { MatSelectModule } from '@angular/material/select';
+import { MatInputModule } from '@angular/material/input';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatDatepickerModule } from '@angular/material/datepicker';
+import { MatNativeDateModule } from '@angular/material/core';
 import { FormControl, FormGroup, Validators } from '@angular/forms';
-
+import { Profile } from '../../shared/models/profile.model';
 
 @Component({
   selector: 'app-info-form',
   templateUrl: './info-form.component.html',
-  styleUrl: './info-form.component.css',
+  styleUrls: ['./info-form.component.css'],
   providers: [
     MatFormFieldModule,
     MatInputModule,
     MatSelectModule,
     MatDatepickerModule,
-    MatNativeDateModule
+    MatNativeDateModule,
   ],
 })
-export class InfoFormComponent {
+export class InfoFormComponent  {
+  @Output() profileChange = new EventEmitter<Profile>();
+  @Input() profile!: Profile;
 
-  infoForm = new FormGroup({ 
+  infoForm = new FormGroup({
     firstName: new FormControl('', [Validators.required]),
     lastName: new FormControl('', [Validators.required]),
-    email: new FormControl('', [Validators.required]),
-    phone: new FormControl('', [Validators.required]),
+    email: new FormControl('', [Validators.required, Validators.email]),
+    phoneNumber: new FormControl('', [Validators.required]),
     address: new FormControl('', [Validators.required]),
-    country: new FormControl('', [Validators.required])
   });
 
+  onFormChange(): void {
+    const updatedProfile: Profile = {
+      name: this.infoForm.get('firstName')?.value || '',
+      surname: this.infoForm.get('lastName')?.value || '',
+      email: this.infoForm.get('email')?.value || '',
+      phoneNumber: this.infoForm.get('phoneNumber')?.value || '',
+      address: this.infoForm.get('address')?.value || '',
+    };
```

Comment:
This code will two-way bind the form to the edit-profile component.
```suggestion
  ngOnChanges(): void {
    if (!this.profile) {
      return;
    }

    this.infoForm.setValue({
      firstName: this.profile.name || '',
      lastName: this.profile.surname || '',
      email: this.profile.email || '',
      phoneNumber: this.profile.phoneNumber || '',
      address: this.profile.address || '',
    });

    this.infoForm.valueChanges.subscribe((value) => {
      this.profile.name = value.firstName || '';
      this.profile.surname = value.lastName || '';
      this.profile.email = value.email || '';
      this.profile.phoneNumber = value.phoneNumber || '';
      this.profile.address = value.address || '';
      this.profileChange.emit(this.profile);
      this.setValid();
    });
  }

  private setValid(): void {
    this.valid = this.infoForm.valid;
    this.validChange.emit(this.valid);
  }
```